web/src/components/merchant/sign-up/SignUpCard.tsx

```diff
@@ -56,31 +65,123 @@ const StyledButton = styled(Button)`
   font-size: 18px;
   font-weight: bolder;
   text-transform: uppercase;
-
-  margin-top: 30px;
+  margin: 0;
 `;
 
 /**
  * This is the card containing the sign up form for the Merchant app.
  * It is displayed in the Sign Up page.
  */
-const SignUpCard: React.FC = () => (
-  <CardContainer>
-    <GroupBuyMerchantHeader />
-    <Form>
-      <FormRow label="Name" inputType="text" />
-      <FormRow label="Email" inputType="email" />
-      <FormRow label="Password" inputType="password" />
-      <FormRow label="Confirm Password" inputType="password" />
-      <FormRow label="VPA" inputType="email" />
+const SignUpCard: React.FC = () => {
```

Comment:
Could you organize all these data accessors/setters/functions of different fields in a custom hook? https://reactjs.org/docs/hooks-custom.html
And then package them into a single UI component as SignUpCard.

Reply:
This is a custom hook, which decouples UI and logic. Since there are too many fields here, I only pick name and email as examples.
```javascript
function useSignupCard() {                                                         
  const [state, setState] = useState({                                             
    name: '',                                                                      
    nameError: '',                                                                 
    email: '',                                                                     
    emailError: ''                                                                 
  });                                                                              
                                                                                   
  const handleEmailChange = (event: React.ChangeEvent<HTMLInputElement>) => {   
    setState(Object.assign(state, {                                                
      email: event.target.value,                                                   
      emailError: event.target.value ? '': EMAIL_EMPTY,                            
    }));                                                                           
  }                                                                                
                                                                                   
  return {                                                                         
    state,                                                                         
    setSate,                                                                       
    handleEmailChange,                                                             
  }                                                                                
}};
```

Reply:
I managed to use react-hook-form library now so most of the logic is done by the library.